extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java

```diff
@@ -603,14 +567,24 @@ void registerGraphQLUiHandler(
             BuildProducer<RouteBuildItem> routeProducer,
             SmallRyeGraphQLRecorder recorder,
             SmallRyeGraphQLRuntimeConfig runtimeConfig,
-            SmallRyeGraphQLBuildItem smallRyeGraphQLBuildItem,
             LaunchModeBuildItem launchMode,
             NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem,
-            SmallRyeGraphQLConfig graphQLConfig) throws Exception {
+            SmallRyeGraphQLConfig graphQLConfig,
+            WebJarResultBuildItem webJarResultBuildItem,
+            BuildProducer<SmallRyeGraphQLBuildItem> smallRyeGraphQLBuildProducer) {
+
+        WebJarResultBuildItem.WebJarResult result = webJarResultBuildItem.byArtifactKey(GRAPHQL_UI_WEBJAR_ARTIFACT_KEY);
+        if (result == null) {
+            return;
+        }
 
         if (shouldInclude(launchMode, graphQLConfig)) {
-            Handler<RoutingContext> handler = recorder.uiHandler(smallRyeGraphQLBuildItem.getGraphqlUiFinalDestination(),
-                    smallRyeGraphQLBuildItem.getGraphqlUiPath(), runtimeConfig);
+            String graphQLUiPath = nonApplicationRootPathBuildItem.resolvePath(graphQLConfig.ui.rootPath);
+            smallRyeGraphQLBuildProducer
```

Comment:
`SmallRyeGraphQLBuildItem` is produced just to be on the safe side, no idea if anyone consumes it.

Maybe we can deprecate the build item as well?